browser/android/brave_shields_content_settings.cc

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/android/brave_shields_content_settings.h"
+
+#include <string>
+
+#include "base/android/jni_android.h"
+#include "base/android/jni_string.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "components/content_settings/core/browser/content_settings_utils.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+
+#include "jni/BraveShieldsContentSettings_jni.h"
+
+static const char kAllowResource[] = "allow";
+
+namespace chrome {
+namespace android {
+
+BraveShieldsContentSettings* _brave_shields_content_settings = nullptr;
+
+static void JNI_BraveShieldsContentSettings_Init(JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jcaller) {
+  _brave_shields_content_settings =
```

Comment:
This seems like an unconventional pattern, typically these methods return `jlong` pointer. Why the global instead?

Reply:
I guess because of this? https://github.com/brave/brave-core/pull/2998/files#diff-9e013f038db0e4190df2d93321fa16a4R65 - but maybe a LazyInstance would be better here like `BackgroundSyncLauncherAndroid`?

Reply:
I think it's almost the same as there. Don't think we need to redo it.